### ISSUE TITLE
Allow to filter tests by pattern

### ISSUE THREAD

**Issue by fesor** [peridot-php/peridot#181]
Currently peridot provide only ability to filter by file. What I propose is additional `--filter` options for CLI which will allows to filter by test case description. For example:

``` php
describe('Some Component', function () {
    it('It works in most common cases (critical)', function () {
         // test case 1
    });

    it('It works in other cases', function () {
        // test case 2
    });
});
```

So by running

```
peridot --filter "\(critical\)" specs/ 
```

We will get only `test case 1` executed. I could prepare a PR but I'm not sure about implementation and maybe this should be just plugin and not built in feature.

**Comment by brianium** [comment]
I had always intended for `--grep` to work this way. The WIP on 2.0 implements this, maybe we could pull it out of that branch? https://github.com/peridot-php/peridot-core/commits/master - the commits starting on June 21 implemented this feature.

**Comment by brianium** [comment]
I kept reaching for the 1.* branch for new projects because its stable and available ¯\\\_(ツ)_/¯

**Comment by fesor** [comment]
Will try to submit PR soon, thanks

**Comment by ezzatron** [comment]
@fesor What do you think of #185? I believe it would probably solve this problem too.

**Comment by ezzatron** [comment]
This functionality will be available next release via new `--focus` and `--skip` options. But perhaps even better, focused specs will allow you to simply prefix `it()` or `describe()` with an `f` (e.g. `fit()`, `fdescribe()`) to temporarily focus on a specific part of your suite.

See #199 for more details.

**Comment by ezzatron** [comment]
Released in [1.19.0](https://github.com/peridot-php/peridot/releases/1.19.0) 🎉
